stablehlo/reference/Element.cpp

```diff
@@ -607,6 +601,203 @@ Element ceil(const Element &el) {
   return Element(el.getType(), val);
 }
 
+Element convert(Type type, bool value) {
+  if (isSupportedBooleanType(type)) return Element(type, value);
+  if (isSupportedIntegerType(type))
+    return Element(type, APInt(type.getIntOrFloatBitWidth(), value ? 1 : 0,
+                               type.cast<IntegerType>().isSigned()));
+  if (isSupportedFloatType(type)) {
+    bool roundingErr;
+    auto floatVal = APFloat(value ? 1.0 : 0.0);
+    floatVal.convert(type.cast<FloatType>().getFloatSemantics(),
+                     APFloat::rmNearestTiesToEven, &roundingErr);
+    return Element(type, floatVal);
+  }
+  if (isSupportedComplexType(type)) {
+    bool roundingErr;
+    auto realVal = APFloat(value ? 1.0 : 0.0);
+    realVal.convert(realVal.getSemantics(), APFloat::rmNearestTiesToEven,
+                    &roundingErr);
+    auto imagVal = APFloat(0.0);
+    imagVal.convert(imagVal.getSemantics(), APFloat::rmNearestTiesToEven,
+                    &roundingErr);
+    return Element(type, std::complex<APFloat>(realVal, imagVal));
+  }
+  report_fatal_error(invalidArgument("Unsupported element type: %s",
+                                     debugString(type).c_str()));
+}
+
+Element convert(Type type, int64_t value) {
+  if (isSupportedBooleanType(type)) return Element(type, value != 0);
+  if (isSupportedIntegerType(type))
+    return Element(
+        type, APInt(type.getIntOrFloatBitWidth(), static_cast<uint64_t>(value),
+                    /*isSigned=*/isSupportedSignedIntegerType(type)));
+  if (isSupportedFloatType(type)) {
+    auto floatVal = APFloat(static_cast<double>(value));
+    bool roundingErr;
+    floatVal.convert(type.cast<FloatType>().getFloatSemantics(),
+                     APFloat::rmNearestTiesToEven, &roundingErr);
+    return Element(type, floatVal);
```

Comment:
Should we maybe use `APFloat::convertFromAPInt` instead? That would avoid the static cast and keep all conversions in the APInt/APFloat domain.